ext/phar/util.c

```diff
@@ -1579,7 +1579,13 @@ int phar_verify_signature(php_stream *fp, size_t end_of_phar, uint32_t sig_type,
 			}
 
 			md_ctx = EVP_MD_CTX_create();
-			EVP_VerifyInit(md_ctx, mdtype);
+			if (!md_ctx || !EVP_VerifyInit(md_ctx, mdtype)) {
+				EVP_MD_CTX_destroy(md_ctx);
```

Comment:
Even though `EVP_MD_CTX_destroy` checks whether `md_ctx` is NULL, it's not assured by docs which states:
```
EVP_MD_CTX_destroy() cleans up digest context ctx and frees up the space allocated to it, it should be called only on a context created using EVP_MD_CTX_create().
```
So to be correct we should also do that check ourseleves...